src/colorizer/ColorizeCanvas.ts

```diff
@@ -459,143 +422,136 @@ export default class ColorizeCanvas {
-      this.setUniform("featureColorRampMax", this.colorMapRangeMax);
-    }
-  }
-
-  setInRangeLUT(inRangeLUT: Uint8Array): void {
+  private setInRangeLUT(inRangeLUT: Uint8Array): void {
     // Save the array to a texture and pass it into the shader
     if (inRangeLUT.length > 0) {
       this.setUniform("inRangeIds", packDataTexture(Array.from(inRangeLUT), FeatureDataType.U8));
-      this.render();
     }
   }
 
-  getColorMapRangeMin(): number {
-    return this.colorMapRangeMin;
-  }
-
-  getColorMapRangeMax(): number {
-    return this.colorMapRangeMax;
-  }
-
   /**
-   * @returns The number of frames in the dataset. If no dataset is loaded,
-   * returns 0 by default.
+   * Forces a reload of the current frame, even if it's already loaded. If a
+   * different frame is in the process of being loaded, the reload will be
+   * called on that pending frame.
    */
-  getTotalFrames(): number {
-    return this.dataset ? this.dataset.numberOfFrames : 0;
+  private forceFrameReload(): void {
+    // Force update on the current frame or the frame that's currently being loaded
+    const frame = this.pendingFrame !== -1 ? this.pendingFrame : this.currentFrame;
+    this.setFrame(frame, true).then(() => {
+      this.render();
+    });
   }
 
-  getCurrentFrame(): number {
-    return this.currentFrame;
+  public setOnFrameChangeCallback(callback: (isMissing: boolean) => void): void {
+    this.onFrameChangeCallback = callback;
   }
 
-  public isValidFrame(index: number): boolean {
-    return index >= 0 && index < this.getTotalFrames();
-  }
+  public setParams(params: CanvasStateParams): void {
+    // TODO: What happens when `setParams` is called again while waiting for a Dataset to load?
+    // May cause visual desync where the color ramp/feature data updates before frames load in fully
+    if (this.params === params) {
+      return;
+    }
+    const prevParams = this.params;
+    this.params = params;
+    const hasDatasetChanged = hasPropertyChanged(params, prevParams, ["dataset"]);
 
-  public setObjectOpacity(percentOpacity: number): void {
-    percentOpacity = Math.max(0, Math.min(100, percentOpacity));
-    this.setUniform("objectOpacity", percentOpacity / 100);
-  }
+    if (hasDatasetChanged) {
+      this.handleNewDataset(params.dataset);
+    }
 
-  public setBackdropSaturation(percentSaturation: number): void {
-    percentSaturation = Math.max(0, Math.min(100, percentSaturation));
-    this.setUniform("backdropSaturation", percentSaturation / 100);
-  }
+    this.updateFeatureData(params.dataset, params.featureKey);
+    this.updateTrackData(params.dataset, params.track);
 
-  public setBackdropBrightness(percentBrightness: number): void {
-    percentBrightness = Math.max(0, Math.min(200, percentBrightness));
-    this.setUniform("backdropBrightness", percentBrightness / 100);
-  }
+    if (hasPropertyChanged(params, prevParams, ["outlierDrawSettings"])) {
```

Comment:
See implementation in `data_utils.ts`. The strings are required to be keys in the property type (`CanvasStateParams`)